crates/consensus/beacon/src/engine/sync.rs

```diff
@@ -276,6 +307,10 @@ where
 pub(crate) enum EngineSyncEvent {
     /// A full block has been downloaded from the network.
     FetchedFullBlock(SealedBlock),
+    /// A range of blocks has been downloaded from the network.
+    /// TODO: dedup with FetchedFullBlock? we may not need it since we can request now with
+    /// count=1, and it will _also_ send both the headers and bodies response right away.
+    FetchedBlocks(Vec<SealedBlock>),
```

Comment:
we don't want this here, instead the engine sync needs to buffer the range responses and yield them one by one with increasing blocknumber

because they will be inserted by the consensus engine and we only want to insert 1 block at a time

same pattern as this for example:
https://github.com/paradigmxyz/reth/blob/5f63a487b809a78e12e6ba9a866218ad61542796/crates/net/network/src/state.rs#L380-L383